control/subscription_group.go

```diff
@@ -405,23 +409,53 @@ func (s *subscriptionGroup) subscribePlugins(id string,
 			"version": plg.Version(),
 			"_block":  "subscriptionGroup.subscribePlugins",
 		}).Debug("plugin subscription")
-		pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
-		if err != nil {
-			serrs = append(serrs, serror.New(err))
-			return serrs
-		}
-		pool.Subscribe(id)
-		if pool.Eligible() {
-			err = s.verifyPlugin(plg)
+		if plg.Details.Uri != nil {
+			// this is a remote plugin
+			pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
+			if pool.Count() < 1 {
+				var resp plugin.Response
+				res, err := http.Get(plg.Details.Uri.String())
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+
+				body, err := ioutil.ReadAll(res.Body)
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+				json.Unmarshal(body, &resp)
+				ap, err := newAvailablePlugin(resp, s.eventManager, nil)
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+				pool.Insert(ap)
```

Comment:
In this if block, the pool count is less than 1, which I think implies that there are no available plugins in the pool which is why you're inserting an available plugin into the pool. 
Won't we also have subscribe the plugin by calling `pool.Subscribe(id)` ?